libbeat/publisher/queue/queue.go

```diff
@@ -34,6 +34,21 @@ type ACKListener interface {
 	OnACK(eventCount int) // number of consecutively published events acked by producers
 }
 
+//Metrics is a set of basic-user friendly metrics that report the current state of the queue. These metrics are meant to be relatively generic and high-level, and when reported directly, can be comprehensible to a user.
```

Comment:
Actually I'd rather _these_ metrics be unambiguous and consistently implementable. Generic / high-level / comprehensible ones should ideally happen in the metrics code in a queue-agnostic way (rather than being duplicated in different queues where the semantics can subtly diverge). So my main comment below is about making the metrics more concrete and low-level 😅

Reply:
Yah, that's a good point, I think. When I started I was super on the fence about how much of the work of "deriving" certain values should be done by the queue, or by the metrics code.